src/main/java/mate/academy/spring/controller/UserController.java

```diff
@@ -0,0 +1,30 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.impl.response.UserResponseMapper;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private final UserService userService;
+    private final UserResponseMapper userResponseMapper;
+
+    public UserController(UserService userService, UserResponseMapper userResponseMapper) {
+        this.userService = userService;
+        this.userResponseMapper = userResponseMapper;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto getUserByEmail(@RequestParam String email) {
+        User user = userService.findByEmail(email)
+                .orElseThrow(() -> new RuntimeException("Does not exist the user in a database"
+                        + " for this email: " + email));
```

Comment:
_Question for a mentor_. Should we throw an exception here? 
In my opinion, this exception should be thrown to the closest part of the program when this exception can appear, what has been done in Dao layer.

Reply:
@viprko  Dao as usual return an Optional value. 
The best practice is when exception is thrown on service layer and service should return an entity (not Optional). But sometimes service method can be called from other service and for business logic requirements should return Optional. So for this task is not a big mistake to throw an exception on Controller layer.